File: 2151-maximum-good-people-based-on-statements/2151-maximum-good-people-based-on-statements.py
class Solution:
    def maximumGood(self, statements: List[List[int]]) -> int:
        def checkGoodCount(mask):
            cnt = 0
            for i in range(n):
                if mask >> i & 1 == 0:
                    continue
                for j in range(n):
                    num = statements[i][j]
                    if num < 2 and num != mask >> j & 1:
                        return False, 0
                cnt += 1
            return True, cnt
          
        
        n, ans = len(statements), 0
        for mask in range(1, 1 << n):
            flag, count = checkGoodCount(mask)
            if flag and count > ans:
                ans = count
        return ans

            
                    

File: 2151-maximum-good-people-based-on-statements/2151-maximum-good-people-based-on-statements.js
/**
 * @param {number[][]} statements
 * @return {number}
 */
var maximumGood = function(statements) {
  const n = statements.length;
  let ans = 0;
  
  for (let mask = 1; mask < 1 << n; mask++) {
    let [flag, cnt] = checkGoodCount(statements, n, mask);
    if (flag && cnt > ans) ans = cnt;
  }
  
  return ans;
};

const checkGoodCount = (statements, n, mask) => {
  let cnt = 0;
  
  for (let i = 0; i < n; i++) {
    if (!(mask >> i & 1)) continue;
    
    for (let j = 0; j < n; j++) {
      const num = statements[i][j];
      if (num < 2 && num !== (mask >> j & 1)) return [false, 0];
    }
    
    cnt++;
  }
  
  return [true, cnt];
}